### PR TITLE
Fix bug, rows count = 0

### PR DESCRIPTION
**Pull request by webrouse**
Jira: https://keboola.atlassian.net/browse/COM-567

Changes:
- Empty table (rows count =0) is now exported to storage.

Old job (`v5.0.0`) before fix:
- https://connection.keboola.com/admin/projects/2703/jobs/698660240
- image

New job, tagged build:
- https://connection.keboola.com/admin/projects/2703/jobs/698656991
- image

### CODE REVIEW

File: src/Extractor/Snowflake.php [hunk 1]
```diff
@@ -167,51 +168,4 @@ protected function getOutputFilename(string $outputTableName): string
         $outTablesDir = $this->dataDir . '/out/tables';
         return $outTablesDir . '/' . $sanitizedTableName . '.csv.gz';
     }
-
-    protected function createManifest(ExportConfig $exportConfig): void
```
Comment (webrouse) [inline src/Extractor/Snowflake.php:171]: Uz nam to netreba, je to vyriesene v kniznici.

File: deploy.sh [hunk 2]
```diff
@@ -1,20 +1,34 @@
 #!/bin/bash
 set -e
 
-docker images
+# Obtain the component repository and log in
```
Comment (webrouse) [inline deploy.sh:4]: Deploy.sh som aktualizoval, pretoze menil v dev portal aj take verzie, ktore nezodpovedali nasmu verziovaniu.

File: src/Extractor/SnowflakeMetadataProvider.php [hunk 3]
```diff
@@ -78,17 +79,27 @@ public function listTables(array $whitelist = [], bool $loadColumns = true): Tab
...
             'SELECT * FROM (%s) LIMIT 0;',
             rtrim(trim($query), ';')
         );
-
         $this->db->query($sql)->fetchAll();
+        $columnsRaw = $this->db->query('DESC RESULT LAST_QUERY_ID()')->fetchAll();
+
+        $columns = [];
+        foreach ($columnsRaw as $data) {
+            [$type, $length] = SnowflakeUtils::parseTypeAndLength($data['type']);
```
Comment (webrouse) [inline src/Extractor/SnowflakeMetadataProvider.php:94]: Vid test.

File: tests/functional/incremental-fetching-empty-rows/expected/data/out/tables/in.c-main.auto-increment-timestamp.csv.gz.manifest [hunk 4]
```diff
@@ -0,0 +1,291 @@
+{
```
Comment (webrouse) [inline tests/functional/incremental-fetching-empty-rows/expected/data/out/tables/in.c-main.auto-increment-timestamp.csv.gz.manifest:1]: Okrem manifestu sa vytvara aj prazdny adresar, vid nizsie `empty-rows/expected/data/out/tables/in.c-main.auto-increment-timestamp.csv.gz/.gitkeep ` ... tj. storage uz tabulku ulozi.

File: tests/phpunit/SnowflakeUtilsTest.php [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        ];
+
+        yield [
+            'NUMBER(38,0)',
+            ['NUMBER', '38,0'],
+        ];
+
+        yield [
+            'VARCHAR(16777216)',
+            ['VARCHAR', '16777216'],
```
Comment (webrouse) [inline tests/phpunit/SnowflakeUtilsTest.php:35]: Test pre parsovanie SNFK typu a jeho `length`.

File: tests/functional/incremental-fetching-empty-rows/expected/data/out/tables/in.c-main.auto-increment-timestamp.csv.gz.manifest (hunk 4 above)
Comment (webrouse) [reply tests/functional/incremental-fetching-empty-rows/expected/data/out/tables/in.c-main.auto-increment-timestamp.csv.gz.manifest:1]: Skusal som to aj v KBC, vid screenshoty v PR popise.

**Review by ondrajodas — APPROVED** [review]
👍

**Pull request merged**